src/Tizen.Security.DevicePolicyManager/Tizen.Security.DevicePolicyManager/StoragePolicy.cs

```diff
@@ -0,0 +1,146 @@
+﻿/*
+ *  Copyright (c) 2018 Samsung Electronics Co., Ltd All Rights Reserved
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License
+ */
+
+using System;
+
+namespace Tizen.Security.DevicePolicyManager
+{
+    /// <summary>
+    /// The StoragePolicy provides methods to manage storage policies.
+    /// </summary>
+    /// <since_tizen> 6 </since_tizen>
+    public class StoragePolicy : DevicePolicy, IDisposable
+    {
+        private readonly string _externalStoragePolicyName = "external_storage";
+        private int _externalStorageCallbackId;
+        private bool _disposed = false;
+
+        private Interop.DevicePolicyManager.PolicyChangedCallback _externalStoragePolicyChangedCallback;
+        private EventHandler<PolicyChangedEventArgs> _externalStoragePolicyChanged;
+
+        internal StoragePolicy()
+        {
+        }
+
+        internal StoragePolicy(DevicePolicyManager dpm) : base(dpm)
+        {
+        }
+
+        /// <summary>
+        /// A Destructor of StoragePolicy.
+        /// </summary>
+        ~StoragePolicy()
+        {
+            this.Dispose(false);
+        }
+
+        /// <summary>
+        /// Releases any unmanaged resources used by this object.
+        /// </summary>
+        public void Dispose()
+        {
+            this.Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <summary>
+        /// Releases any unmanaged resources used by this object. Can also dispose any other disposable objects.
+        /// </summary>
+        /// <param name="disposing">If true, disposes any disposable objects. If false, does not dispose disposable objects.</param>
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!_disposed)
+            {
+                if (disposing)
+                {
+                    // to be used if there are any other disposable objects
+                }
+
+                if (_externalStorageCallbackId != 0)
+                {
+                    try
+                    {
+                        RemoveExternalStoragePolicyChangedCallback();
+                    }
+                    catch (Exception e)
+                    {
+                        Log.Error(Globals.LogTag, e.ToString());
+                    }
+                }
+
+                _disposed = true;
+            }
+        }
+
+        /// <summary>
```

Comment:
Why not provide API to query current policy?

Reply:
I added properties.